compiler/rustc_const_eval/src/interpret/projection.rs

```diff
@@ -201,6 +201,9 @@ where
                 let offset = stride * index; // `Size` multiplication
                 // All fields have the same layout.
                 let field_layout = base.layout().field(self, 0);
+                if field_layout.is_unsized() {
+                    throw_inval!(ConstPropNonsense);
```

Comment:
These should always come with comments explaining what happens and pointers to issues with examples.